sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -2614,5 +2640,259 @@ private void SetNameFieldsIfNull()
+                .ToSyncCollection(cancellationToken);
+
+        /// <summary>
+        /// Gets the paths that have recently been soft deleted in this file system.
+        /// </summary>
+        /// <param name="path">
+        /// Filters results to paths within the specified directory.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// An <see cref="Pageable{PathHierarchyDeletedItem}"/>
+        /// describing the deleted paths in the file system.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual AsyncPageable<PathHierarchyDeletedItem> GetDeletedPathsAsync(
+            string path = default,
+            CancellationToken cancellationToken = default)
+            => new GetDeletedPathAsyncCollection(
+                this,
+                path,
+                $"{nameof(DataLakeFileSystemClient)}.{nameof(GetDeletedPaths)}")
+                .ToAsyncCollection(cancellationToken);
+
+        internal async Task<Response<PathDeletedSegment>> GetDeletedPathsInternal(
+            string path,
+            string continuation,
+            int? maxResults,
+            string operationName,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (ClientConfiguration.Pipeline.BeginLoggingScope(nameof(DataLakeFileSystemClient)))
+            {
+                ClientConfiguration.Pipeline.LogMethodEnter(
+                    nameof(DataLakeFileSystemClient),
+                    message:
+                    $"{nameof(Uri)}: {Uri}\n" +
+                    $"{nameof(continuation)}: {continuation}\n" +
+                    $"{nameof(maxResults)}: {maxResults})");
+
+                operationName ??= $"{nameof(DataLakeFileClient)}.{nameof(GetDeletedPaths)}";
+                DiagnosticScope scope = ClientConfiguration.ClientDiagnostics.CreateScope(operationName);
+
+                try
+                {
+                    scope.Start();
+                    ResponseWithHeaders<ListBlobsHierarchySegmentResponse, FileSystemListBlobHierarchySegmentHeaders> response;
+
+                    if (async)
+                    {
+                        response = await BlobFileSystemRestClient.ListBlobHierarchySegmentAsync(
+                            delimiter: null,
+                            prefix: path,
+                            marker: continuation,
+                            maxResults: maxResults,
+                            include: null,
+                            timeout: null,
+                            cancellationToken: cancellationToken)
+                            .ConfigureAwait(false);
```

Comment:
I find it odd that we're not passing `showOnly` anywhere here but I can see `showOnly` being correctly sent in the recording. How does it work?

Could you add test where we have 1-2 not-deleted files and 1-2 deleted files and make sure normal listing and list deleted return right entries?

Reply:
`showOnly` is hardcoded in swagger.

This API is only used to list deleted paths, normal listing is not an option.  Normal listing is accomplished with `DataLakeFileSystemClient.GetPaths()`.

I can expand the unit test for this API.

Reply:
can showOnly be unhardcoded ? or generated api name changed? this will be confusing to anybody revisiting this code in the future.

Reply:
We will be throwing this all away in less than 6 months when we migrate Data Lake to the Blob endpoint.  When I add this functionality to the Blob swagger, it will not be hardcoded.

Reply:
Please add a comment in the code explaining the situation then.